src/main/java/org/sciborgs1155/lib/SparkUtils.java

```diff
@@ -49,54 +77,88 @@ public static enum Data {
     int status4 = FRAME_STRATEGY_DISABLED; // alternate quadrature encoder | default 20
     int status5 = FRAME_STRATEGY_DISABLED; // duty cycle position | default 200
     int status6 = FRAME_STRATEGY_DISABLED; // duty cycle velocity | default 200
+    int status7 = FRAME_STRATEGY_DISABLED;
+    // status frame 7 is cursed, the only mention i found of it in rev's docs is at
+    // https://docs.revrobotics.com/brushless/spark-flex/revlib/spark-flex-firmware-changelog#breaking-changes
+    // if it's only IAccum, there's literally no reason to enable the frame
+
+    Optional<DoubleSupplier> supplier1 = Optional.empty();
+    Optional<DoubleSupplier> supplier2 = Optional.empty();
+    Optional<DoubleSupplier> supplier3 = Optional.empty();
+    Optional<DoubleSupplier> supplier4 = Optional.empty();
+    Optional<DoubleSupplier> supplier5 = Optional.empty();
+    Optional<DoubleSupplier> supplier6 = Optional.empty();
+
+    if (withFollower || data.contains(Data.APPLIED_OUTPUT)) {
+      status0 = FRAME_STRATEGY_VERY_FAST;
+    }
 
-    if (data.contains(Data.VELOCITY)
-        || data.contains(Data.VOLTAGE)
-        || data.contains(Data.CURRENT)) {
+    if (sensors.contains(Sensor.INTEGRATED) && data.contains(Data.VELOCITY)
+        || data.contains(Data.INPUT_VOLTAGE)
+        || data.contains(Data.CURRENT)
+        || data.contains(Data.TEMPERATURE)) {
       status1 = FRAME_STRATEGY_FAST;
+      supplier1 = Optional.of(spark::getOutputCurrent);
     }
 
-    if (data.contains(Data.POSITION)) {
+    if (sensors.contains(Sensor.INTEGRATED) && data.contains(Data.POSITION)) {
       status2 = FRAME_STRATEGY_FAST;
+      supplier2 = Optional.of(() -> spark.getEncoder().getPosition());
     }
 
-    if (sensors.contains(Sensor.ANALOG)) {
+    if (sensors.contains(Sensor.ANALOG)
+        && (data.contains(Data.VELOCITY) || data.contains(Data.POSITION))) {
       status3 = FRAME_STRATEGY_FAST;
+      supplier3 = Optional.of(() -> spark.getEncoder().getVelocity());
     }
 
-    if (sensors.contains(Sensor.QUADRATURE)) {
+    if (sensors.contains(Sensor.ALTERNATE)
+        && (data.contains(Data.VELOCITY) || data.contains(Data.POSITION))) {
       status4 = FRAME_STRATEGY_FAST;
+      supplier4 =
+          Optional.of(() -> spark.getEncoder(Type.kQuadrature, THROUGHBORE_CPR).getPosition());
     }
 
-    if (sensors.contains(Sensor.DUTY_CYCLE)) {
+    if (sensors.contains(Sensor.ABSOLUTE)) {
       if (data.contains(Data.POSITION)) {
         status5 = FRAME_STRATEGY_FAST;
+        supplier5 = Optional.of(() -> spark.getAbsoluteEncoder().getPosition());
       }
       if (data.contains(Data.VELOCITY)) {
         status6 = FRAME_STRATEGY_FAST;
+        supplier6 = Optional.of(() -> spark.getAbsoluteEncoder().getVelocity());
       }
     }
 
-    if (!withFollower) {
-      status0 = FRAME_STRATEGY_SLOW;
+    int[] frames = {status0, status1, status2, status3, status4, status5, status6, status7};
```

Comment:
the unused stuff here was part of a few jank fixes at nyc, you should probably remove it